Mathlib/RingTheory/Ideal/Basic.lean

```diff
@@ -519,6 +519,24 @@ theorem span_singleton_eq_span_singleton {α : Type u} [CommRing α] [IsDomain 
   apply and_congr <;> rw [span_singleton_le_span_singleton]
 #align ideal.span_singleton_eq_span_singleton Ideal.span_singleton_eq_span_singleton
 
+/-- The equivalence between `Associates R` and the principal ideals of `R` defined by sending the
+class of `x` to the principal ideal generated by `x`. -/
+noncomputable def equivIsPrincipal (R : Type*) [CommRing R] [IsDomain R] :
+    Associates R ≃ {I : Ideal R // Submodule.IsPrincipal I} := by
+  refine Equiv.ofBijective (fun x ↦ ⟨Ideal.span {Quot.out x}, ⟨Quot.out x, rfl⟩⟩)
+    ⟨fun  _ _ h ↦ ?_, fun ⟨I, hI⟩ ↦ ?_⟩
+  · rw [Subtype.mk_eq_mk, span_singleton_eq_span_singleton] at h
+    exact Quotient.out_equiv_out.mp h
+  · obtain ⟨x, hx⟩ := hI
+    refine ⟨⟦x⟧, ?_⟩
+    rw [Subtype.mk_eq_mk, hx, submodule_span_eq, span_singleton_eq_span_singleton]
+    exact Associates.mk_quot_out x
+
+theorem equivIsPrincipal_apply (R : Type*) [CommRing R] [IsDomain R] (x : R) :
```

Comment:
Please change this name as well.
Also, could this be a simp lemma?

Reply:
Yes, good point.